### ISSUE TITLE
Doesn't anyone think the author's center loss is too complicated?

### ISSUE THREAD

**Issue by crj1998** [KaiyangZhou/pytorch-center-loss#20]
A concise and easy to understand version

```
class CenterLoss(nn.Module):
    def __init__(self, num_class=10, num_feature=2):
        super(CenterLoss, self).__init__()
        self.num_class = num_class
        self.num_feature = num_feature
        self.centers = nn.Parameter(torch.randn(self.num_class, self.num_feature))

    def forward(self, x, labels):
        center = self.centers[labels]
        dist = (x-center).pow(2).sum(dim=-1)
        loss = torch.clamp(dist, min=1e-12, max=1e+12).mean(dim=-1)

        return loss
```

**Comment by ajndkr** [comment]
@crj1998 I agree. The above solution is more efficient. I suggest you open a PR for the author.

**Comment by yukilulu** [comment]
Your code is more easy to understand, however, i have no idea that the center of class is where to update, could you help me?

**Comment by RaduFilip16** [comment]
Can you please give an example of how to use this?

**Comment by Anson-He** [comment]
Does anyone know if that works?
